admin/src/app/home/orders/stock-orders/receive/receive.component.ts

```diff
@@ -312,107 +404,150 @@ export class ReceiveComponent implements OnInit, OnDestroy {
         this.loading = false;
-        console.log('error', err);
+        console.log("error", err);
+      }
+    );
+  }
+
+  async searchAndIncrementProduct(sku?: string, force: boolean = false) {
+    this.discrepancyModal.hide();
+    // this.loading = true;
+    this.searchSKUFocused = false;
+    this.selectedCategoryLabelFilter = undefined;
+
+    const products = await productDB.products.toArray();
+
+    // 2. Add a check for the product already exists in indexDB => products.find((item) => item.id === this.order.id)
+    const productDataIfExists = products.find(
+      (products) => products.productModelSku === sku
+    );
+
+    // 3. If product exists in indexDB, then just update the quantity & call API to update the quantity
+    if (productDataIfExists) {
+      // 1 Add API to Queue
+      this.barcodeReceiveService
+        .addToQueue(this.userProfile.orgModelId, this.order.id, sku)
+        .catch((error) => {
+          this.loading = false;
+          this.toastr.error(error.message);
+
+          // 3.3. If API fails, then update the quantity in indexDB
+          this.updateReceivedQuantity({
+            ...productDataIfExists,
+            receivedQuantity: productDataIfExists.receivedQuantity - 1,
+          });
+
+          // 3.2. Update the quantity in indexDB
+          productDB.products.where({ productModelSku: sku }).modify({
+            receivedQuantity: productDataIfExists.receivedQuantity - 1,
+          });
+        });
+
+      // 3.1. Show in UI
+      this.updateReceivedQuantity({
+        ...productDataIfExists,
+        receivedQuantity: productDataIfExists.receivedQuantity + 1,
       });
 
+      // 3.2. Update the quantity in indexDB
+      await productDB.products
+        .where({ productModelSku: sku })
+        .modify({ receivedQuantity: productDataIfExists.receivedQuantity + 1 });
 
+      return;
+    }
+
+    // 4. If item doesn't found in recieving queue, then call API to get the product details
+    this.scanBarcodeAPI(sku);
   }
 
-  searchAndIncrementProduct(sku?: string, force: boolean = false) {
-    this.discrepancyModal.hide()
-    this.loading = true;
-    this.searchSKUFocused = false;
-    this.selectedCategoryLabelFilter = undefined;
-    this.orgModelApi.scanBarcodeStockOrder(this.userProfile.orgModelId,
-      'receive',
-      sku,
-      this.order.id,
-      force)
-      .subscribe((searchedOrderItem) => {
-        if (searchedOrderItem.showDiscrepancyAlert) {
-          this.discrepancyOrderItem = searchedOrderItem;
-          this.discrepancyModal.show()
-        } else {
-          this.toastr.success('Row updated');
-        }
-        this.searchSKUFocused = true;
-        this.receivedLineItems = [searchedOrderItem];
-        this.totalReceivedLineItems = this.receivedLineItems.length;
-        if (!searchedOrderItem.received) {
-          this.notReceivedLineItems = [searchedOrderItem];
-        } else {
+  async scanBarcodeAPI(sku?: string, force: boolean = false) {
```

Comment:
duplicate code 
this code already exists in barcode service 

Can we reuse it  ?

Reply:
Yes. Fixed it.